File: Fixed Stack Dynamic .js
class FixedStackDynamic {
    constructor(size) { this.capacity = size; this.stack = []; }
    push(val) { if(this.stack.length < this.capacity) this.stack.push(val); }
    pop() { return this.stack.length ? this.stack.pop() : -1; }
}

const stack = new FixedStackDynamic(5);
stack.push(10); stack.push(20);
console.log("Popped:", stack.pop());

File: Fixed Stack Dynamic .cpp
#include <iostream>
using namespace std;

class FixedStackDynamic {
    int* stack;
    int top, capacity;
public:
    FixedStackDynamic(int size) : capacity(size), top(-1) {
        stack = new int[capacity];
    }
    void push(int val) { if(top < capacity - 1) stack[++top] = val; }
    int pop() { return (top >= 0) ? stack[top--] : -1; }
    ~FixedStackDynamic() { delete[] stack; }
};

int main() {
    FixedStackDynamic stack(5);
    stack.push(10); stack.push(20);
    cout << "Popped: " << stack.pop() << endl;
    return 0;
}
